File: algorithms/searching/binary-search/BinarySearch.dart
class Searching {
  static binarySearch(List<int> arr, int target) {
    int low = 0;
    int high = arr.length;
    int mid;

    while (low <= high) {
      double half = ((high - low) / 2);
      mid = low + half.toInt();

      if (arr[mid] == target) {
        return mid;
      } else if (arr[mid] > target) {
        high = mid - 1;
      } else {
        low = mid + 1;
      }
    }

    return -1;
  }
}

File: algorithms/searching/binary-search/BinarySearch.java
public class BinarySearch {
 public static int binarySearch(int[] arr, int target) {
    int low = 0;
    int high = arr.length - 1;
    int mid;

    while (high >= low) {
      mid = low + (high - low) / 2;

      if (arr[mid] == target)
        return mid;
      else if (arr[mid] > target)
        high = mid - 1;
      else
        low = mid + 1;
    }

    return -1;
  }
}
